File: src/main/java/br/com/augustoccesar/querybuilder/query/trackers/LimitTracker.java
package br.com.augustoccesar.querybuilder.query.trackers;

import br.com.augustoccesar.querybuilder.builders.Buildable;
import br.com.augustoccesar.querybuilder.query.Limit;

/**
 * Author: augustoccesar
 * Date: 10/05/17
 */
public class LimitTracker implements Buildable {
    private Limit limit;

    public LimitTracker setLimit(int value) {
        this.limit = new Limit(value);
        return this;
    }


    @Override
    public String build() {
        return (" " + this.limit.getValue() + " ").replaceAll("\\s+", " ");
    }

    @Override
    public boolean shouldBuild() {
        return limit != null;
    }
}

File: src/main/java/br/com/augustoccesar/querybuilder/query/trackers/LimitTracker.kt
package br.com.augustoccesar.querybuilder.query.trackers

import br.com.augustoccesar.querybuilder.builders.Buildable
import br.com.augustoccesar.querybuilder.query.Limit

/**
 * Author: augustoccesar
 * Date: 10/05/17
 */
class LimitTracker : Buildable {
    private var limit: Limit? = null

    fun setLimit(value: Int): LimitTracker {
        this.limit = Limit(value)
        return this
    }
    
    override fun build(): String {
        return (" " + this.limit!!.value + " ").replace("\\s+".toRegex(), " ")
    }

    override fun shouldBuild(): Boolean {
        return limit != null
    }
}
